### ISSUE TITLE
Cannot encode/ decode an array of model 

### ISSUE THREAD

**Issue by gloryluu** [hirotakan/MessagePacker#25]
I tried to encode an array of model but I didn't get the right output. Then as a result I cannot decode it as well. Both encode or decode with single model is Ok.
```
struct Chocosaurus: Codable {`
   let _id: String
    let index: Int`
    let guid: String
    let isActive: Bool
    let balance: String
    let picture: String
    let age: Int
    let eyeColor: EyeColor
    let name: String
    let gender: Gender
    let company, email, phone, address: String
    let about, registered: String
    let latitude, longitude: Double
    let tags: [String]
    let friends: [Friend]
    let greeting: String
    let favoriteFruit: FavoriteFruit
    
    enum CodingKeys: String, CodingKey {
        case _id
        case index, guid, isActive, balance, picture, age, eyeColor, name, gender, company, email, phone, address, about, registered, latitude, longitude, tags, friends, greeting, favoriteFruit
    }
}

enum EyeColor: String, Codable {
    case blue = "blue"
    case brown = "brown"
    case green = "green"
}

enum FavoriteFruit: String, Codable {
    case apple = "apple"
    case banana = "banana"
    case strawberry = "strawberry"
}

// MARK: - Friend
struct Friend: Codable {
    let id: Int
    let name: String
}

enum Gender: String, Codable {
    case female = "female"
    case male = "male"
}


func jsonFileToList(fileName fileName: String)-> [Chocosaurus]? {
    if let url = Bundle.main.url(forResource: fileName, withExtension: "json") {
        do {
            return try JSONDecoder().decode([Chocosaurus].self, from: Data(contentsOf: url))
        } catch {
            print("Error!! Unable to parse  \(fileName).json")
        }
    }
    return nil
    
}

func jsonStringToList(jsonString jsonString: String)-> [Chocosaurus]? {
    do {
        return try JSONDecoder().decode([Chocosaurus].self, from:jsonString.data(using: .utf8)!)
    } catch {
        print(error.localizedDescription)
    }
    return nil
    
}

func listToJson(list list: [Chocosaurus])-> String? {
    do {
        let jsonData = try JSONEncoder().encode(list)
        return String(data: jsonData, encoding: .utf8)
    }
    catch {
        print(error.localizedDescription)
    }
    return nil
}

func benchmarks(_ text: String, callback:()->Void) {
    let st = Date().timeIntervalSince1970
    callback()
    let nd = Date().timeIntervalSince1970
    print("\(text) \((nd - st)*1000) ms")
}

let src = jsonFileToList(fileName: "sample")

//// encode to json
let src_j = listToJson(list: src!)

// encode to messagepack
let src_m = try! MessagePackEncoder().encode(src!)

print("JSON: \(src_j!.lengthOfBytes(using: .utf8)) MsgPack: \(src_m.count) Difference: \((1 - src_m.count / src_j!.lengthOfBytes(using: .utf8)) * 100) %")
benchmarks("Encoding with JSON Encoder") { listToJson(list: src!) }
benchmarks("Encoding with msgpack.pack") { try! MessagePackEncoder().encode(src!) }
benchmarks("Decoding with JSON Decoder") { jsonStringToList(jsonString: src_j!) }
//benchmarks("Decoding with msgpack.unpack") { try MessagePackDecoder().decode([Chocosaurus].self, from: src_m) } ///ERROR
```
Sample JSON
```
[
    {
        "_id": "5d78c4fa1c53cced49acfd35",
        "index": 0,
        "guid": "264b7afd-42db-4707-b680-7b01d91e903d",
        "isActive": false,
        "balance": "$1,402.88",
        "picture": "http://placehold.it/32x32",
        "age": 25,
        "eyeColor": "brown",
        "name": "Beth Riddle",
        "gender": "female",
        "company": "DIGITALUS",
        "email": "bethriddle@digitalus.com",
        "phone": "+1 (873) 595-3292",
        "address": "515 Aster Court, Darlington, Wisconsin, 4063",
        "about": "Sint sint voluptate Lorem dolor elit sint dolore proident cillum laborum laborum commodo magna. Et reprehenderit sit anim nostrud elit qui Lorem esse aliqua fugiat ullamco dolor cupidatat id. Proident officia in velit deserunt enim velit officia nisi. Tempor fugiat voluptate magna elit ullamco ullamco. Laborum sint eiusmod ullamco reprehenderit adipisicing reprehenderit elit deserunt incididunt.\r\n",
        "registered": "2018-11-28T10:44:07 -07:00",
        "latitude": -8.17012,
        "longitude": -40.725295,
        "tags": [
            "quis",
            "enim",
            "ut",
            "proident",
            "do",
            "ut",
            "anim"
        ],
        "friends": [
            {
                "id": 0,
                "name": "Sosa Miles"
            },
            {
                "id": 1,
                "name": "Tucker Rowland"
            },
            {
                "id": 2,
                "name": "James Huff"
            }
        ],
        "greeting": "Hello, Beth Riddle! You have 3 unread messages.",
        "favoriteFruit": "strawberry"
    },
    {
        "_id": "5d78c4fa8cee3b66fdd56283",
        "index": 1,
        "guid": "782ef1d7-4dfc-4db7-b32e-ab43e5b33d3e",
        "isActive": false,
        "balance": "$1,060.44",
        "picture": "http://placehold.it/32x32",
        "age": 40,
        "eyeColor": "brown",
        "name": "York Benton",
        "gender": "male",
        "company": "NETPLAX",
        "email": "yorkbenton@netplax.com",
        "phone": "+1 (846) 486-2041",
        "address": "736 Desmond Court, Brewster, Virginia, 4140",
        "about": "Eu id consectetur anim ullamco enim. Minim veniam veniam Lorem ipsum elit fugiat enim proident qui eu commodo in qui. Deserunt eiusmod enim occaecat est ad. Enim duis et laboris est consequat. Nostrud aliquip aliqua labore laborum et officia ex labore laboris elit ipsum non. Excepteur consequat consectetur in in. Sunt dolore velit exercitation aute in culpa non minim incididunt aliquip dolore laborum amet.\r\n",
        "registered": "2015-08-30T05:52:25 -07:00",
        "latitude": 31.973032,
        "longitude": -78.509382,
        "tags": [
            "tempor",
            "nisi",
            "mollit",
            "dolore",
            "eiusmod",
            "amet",
            "tempor"
        ],
        "friends": [
            {
                "id": 0,
                "name": "Guerrero Webster"
            },
            {
                "id": 1,
                "name": "Hopkins Conley"
            },
            {
                "id": 2,
                "name": "Concetta Barry"
            }
        ],
        "greeting": "Hello, York Benton! You have 1 unread messages.",
        "favoriteFruit": "banana"
    }
]
```

**Comment by hirotakan** [comment]
I found a map encoding problem.
I fixed it with this [PR](https://github.com/hirotakan/MessagePacker/pull/26) and released [v0.4.1](https://github.com/hirotakan/MessagePacker/releases/tag/0.4.1).

Please try a new version.

Thank you.

**Comment by gloryluu** [comment]
It is fixed. Thank you.
